src/main/java/org/sagebionetworks/EvaluationUtils.java

```diff
@@ -142,6 +168,84 @@ public static boolean isBooleanAnnotationPresentAndTrue(SubmissionStatus status,
+			annotations.getAnnotations().put(key, annotationsValue);
+		} else {
+			AnnotationsValue annotationsValue = new AnnotationsValue();
+			annotationsValue.setType(AnnotationsValueType.STRING);
+			List<String> stringList = new ArrayList<>();
+			stringList.add(value);
+			annotationsValue.setValue(stringList);
+			annotations.getAnnotations().put(key, annotationsValue);
+		}
+	}
+
+	public static void setAnnotationLongV2(SubmissionStatus status, String key, Long value, boolean isPrivate) {
+		org.sagebionetworks.repo.model.annotation.v2.Annotations annotations = status.getSubmissionAnnotations();
+		if (annotations == null) {
+			annotations = new org.sagebionetworks.repo.model.annotation.v2.Annotations();
+			status.setSubmissionAnnotations(annotations);
+		}
+		if (annotations.getAnnotations() == null) {
+			annotations.setAnnotations(new HashMap<String, AnnotationsValue>());
+		}
+
+		if (annotations.getAnnotations().containsKey(key)) {
+			AnnotationsValue annotationsValue  = annotations.getAnnotations().get(key);
+			annotationsValue.setType(AnnotationsValueType.LONG);
+			annotationsValue.getValue().clear();
+			annotationsValue.getValue().add(value.toString());
+			annotations.getAnnotations().put(key, annotationsValue);
+		} else {
+			AnnotationsValue annotationsValue = new AnnotationsValue();
+			annotationsValue.setType(AnnotationsValueType.LONG);
+			List<String> stringList = new ArrayList<>();
+			stringList.add(value.toString());
+			annotationsValue.setValue(stringList);
+			annotations.getAnnotations().put(key, annotationsValue);
+		}
+	}
+
+	public static void setAnnotationDoubleV2(SubmissionStatus status, String key, Double value, boolean isPrivate) {
+		org.sagebionetworks.repo.model.annotation.v2.Annotations annotations = status.getSubmissionAnnotations();
+		if (annotations == null) {
+			annotations = new org.sagebionetworks.repo.model.annotation.v2.Annotations();
+			status.setSubmissionAnnotations(annotations);
+		}
+		if (annotations.getAnnotations() == null) {
+			annotations.setAnnotations(new HashMap<String, AnnotationsValue>());
+		}
+
+		if (annotations.getAnnotations().containsKey(key)) {
+			AnnotationsValue annotationsValue  = annotations.getAnnotations().get(key);
+			annotationsValue.setType(AnnotationsValueType.DOUBLE);
+			annotationsValue.getValue().clear();
+			annotationsValue.getValue().add(value.toString());
+			annotations.getAnnotations().put(key, annotationsValue);
+		} else {
+			AnnotationsValue annotationsValue = new AnnotationsValue();
+			annotationsValue.setType(AnnotationsValueType.DOUBLE);
+			List<String> stringList = new ArrayList<>();
+			stringList.add(value.toString());
+			annotationsValue.setValue(stringList);
+			annotations.getAnnotations().put(key, annotationsValue);
+		}
+	}
+
 	public static void setAnnotation(SubmissionStatus status, String key, String value, boolean isPrivate) {
 		if (value!=null && value.length()>499) value = value.substring(0, 499);
```

Comment:
This check needs to be replicated for annotations V2 if there is a good reason for it (e.g. is this a limit of the length of the annotation value?).